### ISSUE TITLE
Javascript exception with Blazor Wasm .Net 5 RC 1

### ISSUE THREAD

**Issue by bancroftway** [Tewr/BlazorWorker#25]
I am trying to use BlazorWorker in .Net 5 RC 1.

I am getting this exception in browser developer tools:

```
Uncaught (in promise) DOMException: Failed to execute 'importScripts' on 'WorkerGlobalScope': The script at 'https://localhost:44365/_framework/wasm/dotnet.5.0.0-rc.1.20451.14.js' failed to load.
    at blob:https://localhost:44365/4eab1338-d80a-4c4e-b704-aa7277d64a7c:140:22
```

Here is my code that I am trying to execute on the web-worker:

```
// Create worker.
var worker = await workerFactory.CreateAsync();

// Create service reference. For most scenarios, it's safe (and best) to keep this 
// reference around somewhere to avoid the startup cost.
var service = await worker.CreateBackgroundServiceAsync<CalculationService>();

// Reference that live outside of the current scope should not be passed into the expression.
 // To circumvent this, create a scope-local variable like this, and pass the local variable.
var localRigVariable = appState.CurrentRigBeingEdited;
var localReferenceDataVariable = appState.ReferenceDataQueryResponse;
var result = await service.RunAsync(s => s.RecalculateActivities(localRigVariable, localReferenceDataVariable));
```

**Comment by Tewr** [comment]
The blazor js folder might have changed in 5rc1. I just read the filename from blazorboot, but there is no directory information in there iirc, so the path is hard-coded.

**Comment by bancroftway** [comment]
@Tewr Could you please attempt to upgrade the sample to .Net 5 and advise what changes are to be made? Thanks

**Comment by marcussacana** [comment]
why support a not stable release?
I know how the .net 5 is promising but if they change again in the last release will just break again soon, let's wait some few days.

**Comment by Tewr** [comment]
If I release a package I'll do pre-release and tag it 5rc1.

However, I spent some time updating the project yesterday.  yes they changed the paths but a lot more has changed. lot of runtime problems, so won't be able to release immediately, not a quick update.

**Comment by Epictek** [comment]
@Tewr any ETA when you might have something? I understand that it's a lot of work. Thank you for your efforts :)

**Comment by Tewr** [comment]
no ETA yet, but I can give a status update. I've managed to get up .NET and running in the worker process, but not WebAssembly.Bindings as it's not compiled against .NET5 and won't load. So I either have to find the source code of that dll, or try to decompile-recompile the bytecode against .NET5. Another more sane alternative would be loading the blazor dlls like @marcussacana  did in the past when that thing was broken. But its not as easy as back then, because it now need additional blazor javascript to work, and loading the entire blazor js probably wont fly in the worker context, not to mention it's a huge bloat of code to load into memory just to do some interop. But for now I'm trying something down that road...

**Comment by Epictek** [comment]
Here's the source code for webassembly.binding, hope it helps https://github.com/mono/mono/tree/master/sdks/wasm/framework/src/WebAssembly.Bindings

**Comment by Tewr** [comment]
I have spent a few more minutes on this and apparently the method that was called to get a js object reference that was used by `WebAssembly.Bindings` no longer exists -- or at least it wont bind like it did before. It might be bound to a static memory address, so maybe a specific version. I'm not an expert on internal calls.... I'm trying to call the following method, that looks like it's defined in the current version  here: https://github.com/mono/mono/blob/2966c71ebf2445a5d18e5865ee6728b30f260e74/sdks/wasm/src/driver.c#L353

this is the exception that I get:
```
WASM-WORKER: Your mono runtime and class libraries are out of sync.
WASM-WORKER: The out of sync library is: data-0x2c15238
WASM-WORKER: 
WASM-WORKER: When you update one from git you need to update, compile and install
WASM-WORKER: the other too.
WASM-WORKER: Do not report this as a bug unless you're sure you have updated correctly:
WASM-WORKER: you probably have a broken mono install.
WASM-WORKER: If you see other errors or faults after this message they are probably related
WASM-WORKER: and you need to fix your mono install first.

Uncaught (in promise) Error: System.MissingMethodException:  assembly:<unknown assembly> type:<unknown type> member:(null)
   at WebAssembly.Runtime.InvokeJS(String str)
   at BlazorWorker.WorkerCore.WebAssemblyBindingsProxy.Runtime.InvokeJS(String js)
(...)

```
Same error if I use the nuget version or if I recompile WebAssembly.Runtime against .NET5, so I dont think this is the actual problem, or at least recompiling is far from the solution. But I can't figure out what changed in .NET5, the driver looks quite similar. So alas, something is missing. Probably some fundamental understanding of this code on my part. 

I'll have to get a second look on using the blazor dlls for now.

**Comment by Epictek** [comment]
https://github.com/unoplatform/Uno.Wasm.Bootstrap/blob/dev/jela/net5/src/Uno.Wasm.Bootstrap/WasmScripts/uno-bootstrap.js

Might give some hints?

In looking at Uno I discovered that it supports multithreading, so maybe Blazor support for true Multithreading is not too far away?!
https://platform.uno/blog/webassembly-threading-in-net/

**Comment by Tewr** [comment]
Rumor has it, planned for .net 6 

https://github.com/dotnet/aspnetcore/issues/17730#issuecomment-718006037

**Comment by Tewr** [comment]
Mostly some notes for myself, but I've made a small breakthrough: I've found the .NET5 version of `WebAssembly.Bindings` -- it's hidden inside `System.Private.Runtime.InteropServices.JavaScript`. All in theory, but what should be left is re-routing calls to this assembly instead, and then start cleaning up the dependency graph, as currently I've brute-forced the runtime by loading the entire contents of the blazor bootloader.

**Comment by ivan-prodanov** [comment]
Looking forward to this :)

**Comment by Tewr** [comment]
I have a working version but its currently not deliverable. I need a few more hours to push a release, bear with me.

**Comment by adambajguz** [comment]
When do you plan a release? I do really need your library for my project ;)

**Comment by ivan-prodanov** [comment]
@adambajguz clone the .net 5 branch, it is working.

**Comment by adambajguz** [comment]
@ivan-prodanov  ok

**Comment by Tewr** [comment]
Fixed in [release 2.0.0](https://github.com/Tewr/BlazorWorker/releases/tag/v2.0.0)

**Comment by rootflood** [comment]
hi I had same problem with WASM and it fixed
you can use https://github.com/monsajem/Incs/tree/main/Others/WASM for better workaround
